config/foreman.migrations/20200305172758_enable_puma.rb

```diff
@@ -0,0 +1,5 @@
+unless answers['foreman']['ipa_authentication']
+    unless answers['foreman']['keycloak']
+      answers['foreman']['passenger'] = false if answers['foreman'].is_a?(Hash) && answers['foreman']['passenger']
+    end
+end
```

Comment:
At this point it's not guaranteed that `answers['foreman']` is a hash and might result in `nil`. Then you have `nil['ipa_authentication']` which results in an error.

You could write it like:
```suggestion
if answers['foreman'].is_a?(Hash)
  mod_answers = answers['foreman']
  if mod_answers['passenger'] && !(mod_answers['ipa_authentication'] || mod_answers['keycloak'])
    answers['foreman']['passenger'] = false
  end
end
```

Reply:
good point about `answers['foreman']`

Reply:
@ekohl updated now, thanks